internal/coreinternal/processor/filterspan/filterspan.go

```diff
@@ -111,7 +111,11 @@ func SkipSpan(include Matcher, exclude Matcher, span pdata.Span, resource pdata.
 func (mp *propertiesMatcher) MatchSpan(span pdata.Span, resource pdata.Resource, library pdata.InstrumentationLibrary) bool {
 	// If a set of properties was not in the mp, all spans are considered to match on that property
 	if mp.serviceFilters != nil {
-		serviceName := serviceNameForResource(resource)
+		// Check resource and spans for service.name
+		serviceName, found := serviceNameForResource(resource)
```

Comment:
Is this related to another issue?

Reply:
I see github lost my comment, Yes Datadog does not conform to spec so you ahve to check both places